src/OpenTK.NT/Native/Gdi32/Structs/PixelFormatDescriptor.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System.Runtime.InteropServices;
+
+using BYTE = System.Byte;
+using DWORD= System.UInt32;
+using WORD = System.UInt16;
+
+namespace OpenTK.NT.Native
+{
+    /// <summary>
+    /// Describes a pixel format. It is used when interfacing with the WINAPI to create a new Context.
+    /// Found in WinGDI.h
+    /// </summary>
+    [StructLayout(LayoutKind.Sequential)]
+    public struct PixelFormatDescriptor
+    {
+        public WORD Size;
+        public WORD Version;
+        public PixelFormatDescriptorFlags Flags;
+        public PixelFormatDescriptorPixelTypes PixelType;
+        public BYTE ColorBits;
+        public BYTE RedBits;
+        public BYTE RedShift;
+        public BYTE GreenBits;
+        public BYTE GreenShift;
+        public BYTE BlueBits;
+        public BYTE BlueShift;
+        public BYTE AlphaBits;
+        public BYTE AlphaShift;
+        public BYTE AccumBits;
+        public BYTE AccumRedBits;
+        public BYTE AccumGreenBits;
+        public BYTE AccumBlueBits;
+        public BYTE AccumAlphaBits;
+        public BYTE DepthBits;
+        public BYTE StencilBits;
+        public BYTE AuxBuffers;
+        public PixelFormatDescriptorLayerTypes LayerType;
+        public BYTE Reserved;
+        public DWORD LayerMask;
+        public DWORD VisibleMask;
+        public DWORD DamageMask;
+
+        public static readonly uint SizeInBytes = (uint)Marshal.SizeOf<PixelFormatDescriptor>();
```

Comment:
The Size field should be set to this value, as per the MSDN documentation. https://msdn.microsoft.com/en-us/library/windows/desktop/dd368826(v=vs.85).aspx

Reply:
I know, but I don't see an easy way of doing this without turning this `struct` into a class.

Constructors have to assign to all fields, and `public readonly WORD Size = SizeInBytes;` is not allowed either.